bokehjs/test/unit/models/tools/edit/poly_edit_tool.ts

```diff
@@ -245,11 +253,15 @@ describe("PolyEditTool", (): void => {
       testcase.draw_tool_view._pan_end(pan_event)
 
       expect(testcase.vertex_source.selected.indices).to.be.equal([])
-      expect(testcase.vertex_source.data.x).to.be.equal([0, 0.4646017699115044, 1])
-      expect(testcase.vertex_source.data.y).to.be.equal([0, -0.4661016949152542, -1])
-      expect(testcase.data_source.data.xs).to.be.equal([[0, 0.5, 1], [0, 0.4646017699115044, 1]])
-      expect(testcase.data_source.data.ys).to.be.equal([[0, -0.5, -1], [0, -0.4661016949152542, -1]])
-      expect(testcase.data_source.data.z).to.be.equal([null, null])
+      expect(testcase.vertex_source.data).to.be.equal({
+        x: [0, 0.4646017699115044, 1],
+        y: [0, -0.4661016949152542, -1],
+      })
+      expect(testcase.data_source.data).to.be.equal({
+        xs: [[0, 0.5, 1], [0, 0.4646017699115044, 1]],
+        ys: [[0, -0.5, -1], [0, -0.4661016949152542, -1]],
+        z: [null, null],
+      })
```

Comment:
Why is this form preferable? Presumably split out as they were before affords finer-grained test failure messages

Reply:
I prefer to have full context of a data structure, to be able to see broader failure pattern, which testing one line at a time doesn't allow. It also forces the writer to deal with all fields and there were some missing in other tests.  It would be optimal to pair this with adding support for data structure diff to bokehjs' testing framework, which I have intended to add for a long while.